src/formatters/Utils.cc

```diff
@@ -32,132 +32,14 @@
-    }
-    if ((fpaExposureId & 0xfffffffe00000000LL) != 0LL) {
-        throw LSST_EXCEPT(ex::RangeError, "\"fpaExposureId\" is too large");
-    }
-    return fpaExposureId;
-}
-
-int extractCcdId(std::shared_ptr<PropertySet const> const& properties) {
-    if (properties->isArray("ccdId")) {
-        throw LSST_EXCEPT(ex::RuntimeError, "\"ccdId\" property has multiple values");
-    }
-    int ccdId = properties->getAsInt("ccdId");
-    if (ccdId < 0) {
-        throw LSST_EXCEPT(ex::RangeError, "negative \"ccdId\"");
-    }
-    if (ccdId > 255) {
-        throw LSST_EXCEPT(ex::RangeError, "\"ccdId\" is too large");
-    }
-    return static_cast<int>(ccdId);
-}
-
-int extractAmpId(std::shared_ptr<PropertySet const> const& properties) {
-    if (properties->isArray("ampId")) {
-        throw LSST_EXCEPT(ex::RuntimeError, "\"ampId\" property has multiple values");
-    }
-    int ampId = properties->getAsInt("ampId");
-    if (ampId < 0) {
-        throw LSST_EXCEPT(ex::RangeError, "negative \"ampId\"");
-    }
-    if (ampId > 63) {
-        throw LSST_EXCEPT(ex::RangeError, "\"ampId\" is too large");
-    }
-    return (extractCcdId(properties) << 6) + ampId;
-}
-
-int64_t extractCcdExposureId(std::shared_ptr<PropertySet const> const& properties) {
-    if (properties->isArray("ccdExposureId")) {
-        throw LSST_EXCEPT(ex::RuntimeError, "\"ccdExposureId\" property has multiple values");
-    }
-    int64_t ccdExposureId = properties->getAsInt64("ccdExposureId");
-    if (ccdExposureId < 0) {
-        throw LSST_EXCEPT(ex::RangeError, "negative \"ccdExposureId\"");
-    }
-    return ccdExposureId;
-}
-
-int64_t extractAmpExposureId(std::shared_ptr<PropertySet const> const& properties) {
-    if (properties->isArray("ampExposureId")) {
-        throw LSST_EXCEPT(ex::RuntimeError, "\"ampExposureId\" property has multiple values");
-    }
-    int64_t ampExposureId = properties->getAsInt64("ampExposureId");
-    if (ampExposureId < 0) {
-        throw LSST_EXCEPT(ex::RangeError, "negative \"ampExposureId\"");
-    }
-    return ampExposureId;
-}
-
-std::string const getItemName(std::shared_ptr<PropertySet const> const& properties) {
-    if (!properties) {
-        throw LSST_EXCEPT(ex::InvalidParameterError, "Null std::shared_ptr<PropertySet>");
-    }
-    if (properties->isArray("itemName")) {
-        throw LSST_EXCEPT(ex::InvalidParameterError, "\"itemName\" property has multiple values");
-    }
-    return properties->getAsString("itemName");
-}
-
-bool extractOptionalFlag(std::shared_ptr<PropertySet const> const& properties, std::string const& name) {
-    if (properties && properties->exists(name)) {
-        return properties->getAsBool(name);
-    }
-    return false;
-}
-
-int countFitsHeaderCards(lsst::daf::base::PropertySet const& prop) { return prop.paramNames(false).size(); }
-
 ndarray::Array<std::uint8_t, 1, 1> stringToBytes(std::string const& str) {
```

Comment:
Obvious question is whether to move what's left of this file into another location because it's not really anything to do with `formatters` any more.

Reply:
Seems like the place to move it would be somewhere under afw.table.  It's not really as connected with typehandling.

Reply:
I see that these two functions are only used for tables so I suppose I could try to move them somewhere. Unless you have a good idea where I'm inclined to merge and get the easy win of daf_persistence avoidance.

Reply:
I don't understand.  If you've removed them, but they are still "used for tables", why isn't something breaking?

Reply:
We are talking about the two functions that are left in the file. They seem to be used by WCS for example to convert to bytes to stuff into a table and then convert back to strings to read out of a table.

Reply:
Ah, ok.  I'd lean towards moving them into the source files that use them, if it's just a couple of source files (not headers), but if it's much work at all, then leaving them where they are is probably best, since reducing churn is good, too.

Reply:
No. There are four places:
```
include/lsst/afw/typehandling/python.h
178:        std::string stringRep = formatters::bytesToString(record.get(keys.bytes));
202:    record->set(keys.bytes, formatters::stringToBytes(o));

src/math/TransformBoundedField.cc
112:        auto frameSetStr = formatters::bytesToString(record.get(keys.frameSet));
137:    record->set(keys.frameSet, formatters::stringToBytes(getTransform().writeString()));

src/geom/SkyWcs.cc
98:        std::string stringRep = formatters::bytesToString(record.get(keys.wcs));
385:    record->set(keys.wcs, formatters::stringToBytes(writeString()));

src/geom/Transform.cc
232:        std::string stringRep = formatters::bytesToString(record.get(keys.bytes));
244:    record->set(keys.bytes, formatters::stringToBytes(writeString()));
```

Reply:
So I'll merge.

Reply:
I think the right thing to do might have been to look at the common usage in the above four locations.  It's always getting or setting in a BaseRecord, so that might be the place to move it.